src/DesktopAnalytics/Analytics.cs

```diff
@@ -357,10 +378,18 @@ public static void IdentifyUpdate(UserInfo userInfo)
 		}
 
 		/// <summary>
-		/// Override this for any reason you like, including if the built-in one (http://ipecho.net/plain) stops working some day.
-		/// The service should simply return a page with a body containing the ip address alone.
+		/// Override this if you want your analytics to report an actual IP address to the server (which could be considered PII), rather than
+		/// just the general geolocation info. The service should simply return a page with a body containing the ip address alone.
 		/// </summary>
+		/// <remarks>This used to default to "http://icanhazip.com"; //formerly: "http://ipecho.net/plain" (that URL went down, but is now back up)</remarks>
 		public static string UrlThatReturnsExternalIpAddress { get; set; }
+		/// <summary>
+		/// Override this for any reason you like, including if the built-in one ( http://ip-api.com/json/) stops working some day.
+		/// This will be ignored if <seealso cref="UrlThatReturnsExternalIpAddress"/> is set.
+		/// The service should return json that contains values for one or more of the following names: city, country, countryCode,
+		/// region, regionName.
+		/// </summary>
+		public static string UrlThatReturnsGeolocationJson { get; set; }
```

Comment:
Not news to you, but the problem here is that unlike ip, the json codes may well not match. I was looking at our current default here, and SSL is a paid option. It seems likely that HTTP could cause problems in the future. So I found https://ipstack.com/, but it, for example, has "country_code". (it also does not give away SSL, btw).  Ah well.

Reply:
So if we were serious about making his configurable, we could change this to a method that takes a url and then 3 or 4 parameters corresponding to the json names to get at country, city, etc.

Reply:
Yeah, actually I didn't really think that far ahead. My slightly-more-configurable option was a response to the other URLs keeling over. In reality, if the current solution stops working, we'll probably want to fix it, but by letting the client configure it, they aren't stuck waiting on us, and they might be able to overcome some challenge specific to their needs. Do you think I should go ahead and make it more fully configurable?

Reply:
No, because really, "they" are just you and me :-)

Reply:
It's on nuget. There could be a million users... :-)